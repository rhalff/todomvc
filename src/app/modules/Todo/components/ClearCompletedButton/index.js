// @flow
import React, { PureComponent } from 'react'
import styles from './scss/ClearCompletedButton.scss'

type Props = {
  /**
   * Callback triggered when
   */
  onClear: Function
}

class ClearCompletedButton extends PureComponent<Props> {
  onClearCompleted = () => {
    this.props.onClear()
  }

  render() {
    return (
      <button
        className={styles.clearCompletedButton}
        onClick={this.onClearCompleted}
      >
        Clear completed
      </button>
    )
  }
}

export default ClearCompletedButton
