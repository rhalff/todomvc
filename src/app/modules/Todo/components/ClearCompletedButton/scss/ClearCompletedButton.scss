.clearCompletedButton {
	line-height: 20px;
	text-decoration: none;
	cursor: pointer;
  // should have same style
  // &:active

  &:hover {
	  text-decoration: underline;
  }
}

