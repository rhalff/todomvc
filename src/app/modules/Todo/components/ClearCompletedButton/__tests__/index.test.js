// @flow
import React from 'react'
import ClearCompletedButton from '../'
import renderer from 'react-test-renderer'

test('Can show clear completed', () => {
  const component = renderer.create(<ClearCompletedButton onClear={() => {}} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
