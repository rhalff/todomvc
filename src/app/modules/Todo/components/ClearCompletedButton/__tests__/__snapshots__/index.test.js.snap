// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show clear completed 1`] = `
<button
  className="clearCompletedButton"
  onClick={[Function]}
>
  Clear completed
</button>
`;
